backend/franky/service/dockerhub/dockerhub.py

```diff
@@ -0,0 +1,172 @@
+                            tags[tag] = tags.get(tag, 0) + 1
+            if not response_data['next']:
+                break
+            else:
+                payload['page'] += 1
+        return tags
+
+    def projects(self, username: str) -> List[ProjectData]:
+        username = user_cleaner(username)
+        request_url = self._api_url('%s/%s' % (self._repository_url, username))
+        payload = {'page_size': self._page_size, 'page': 1, 'ordering': 'last_updated'}
+        datas = []
+        while True:
+            response_data = self._call(request_url, payload)
+            repositories = response_data['results']
+            for repository in repositories:
+                image_name = repository['name']
+                image = self._get_image(username, image_name)
+                raw_creation_date = size = tags = None
+                if image:
+                    raw_creation_date = self._parse_datetime(image['creation_date'])
+                    size = image['size']
+                    tags = list(image['tags'])
+                raw_latest_push_date = self._parse_datetime(repository['last_updated'])
+                if raw_latest_push_date + self._active_projects_period > datetime.utcnow():
+                    raw_latest_push_date = None
+                downloads = repository['pull_count']
+                stars = repository['star_count']
+                url_to_repository = self._get_url_repository(username, image_name)
+                datas.append(
+                    ProjectData(name=image_name, start=raw_creation_date, end=raw_latest_push_date, tags=tags,
+                                url=url_to_repository, size=size, pull_count=downloads, star_count=stars))
+            if not response_data['next']:
+                break
+            else:
+                payload['page'] += 1
+        return datas
+
+    def _get_image(self, username, image_name: str) -> Optional[Dict]:
+        request_url = self._api_url('%s/%s/%s/%s' % (self._repository_url, username, image_name, self._tag_url))
+        payload = {'page_size': self._page_size, 'page': 1, 'ordering': 'last_updated'}
+        datas = []
+        tags_list = set()
+        while True:
+            response_data = self._call(request_url, payload)
+            results = response_data['results']
+            for result in results:
+                images = result['images']
+                for image in images:
+                    for index in self._tag_indexes:
+                        if index in image and image[index]:
+                            tags_list.add(image[index])
+                datas.append({'last_updated': result['last_updated'], 'size': result['full_size']})
+            if not response_data['next']:
+                break
+            else:
+                payload['page'] += 1
+        if datas:
+            return {'creation_date': datas[-1]['last_updated'], 'size': datas[0]['size'], 'tags': tags_list}
```

Comment:
Why do you use the latest image tag update date and the first image tag size? I support size should be taken from the latest image as well.

Reply:
For size used latest image and for creation_date used first image

Reply:
Oh, my bad :)